test/testdrive/csv-sources.td

```diff
@@ -16,200 +18,234 @@ place""",CA,92679
 New York,NY,10004
 
 > SELECT * FROM dynamic_csv
-city           state     zip     mz_line_no
---------------------------------------------
+city           state     zip     mz_offset
+------------------------------------------
 Rochester      NY        14618   1
 "New York"     NY        10004   2
 
-$ file-append path=deleting.csv
-city,state,zip
-Tucson,AZ,85719
-
-> CREATE SOURCE deleting_csv
-  FROM FILE '${testdrive.temp-dir}/deleting.csv'
-  FORMAT CSV WITH HEADER
-
-$ file-delete path=deleting.csv
-
-> CREATE INDEX i ON deleting_csv(city,state,zip)
-
-! SELECT * FROM deleting_csv
-regex:Source error: .*: file IO: file source: unable to open file at path
-
-! CREATE SOURCE should_fail FROM FILE '/'
-contains:/ is a directory.
-
 # Static malformed CSV
-$ file-append path=malformed.csv
-Dollars,Category
+$ s3-put-object bucket=test key=malformed.csv
+dollars,category
 5161669,Clothing&Shoes
 1000000000
 ,badrow
 badint,
 
 > CREATE MATERIALIZED SOURCE malformed_csv
-  FROM FILE '${testdrive.temp-dir}/malformed.csv'
-  FORMAT CSV WITH HEADER
+  FROM S3 DISCOVER OBJECTS MATCHING 'malformed.csv' USING BUCKET SCAN 'testdrive-test-${testdrive.seed}'
+  WITH (
+    region = '${testdrive.aws-region}',
+    endpoint = '${testdrive.aws-endpoint}',
+    access_key_id = '${testdrive.aws-access-key-id}',
+    secret_access_key = '${testdrive.aws-secret-access-key}',
+    token = '${testdrive.aws-token}'
+  )
+  FORMAT CSV WITH HEADER (dollars, category)
 
 ! SELECT * FROM malformed_csv
 contains:Decode error: Text: CSV error at record number 3: expected 2 columns, got 1.
 
 # Static non-utf-8 CSV
-$ file-append path=bad-text.csv
-Dollars,Category
+$ s3-put-object bucket=test key=bad-text.csv
+dollars,category
 5161669,\x80
 
 > CREATE MATERIALIZED SOURCE bad_text_csv
-  FROM FILE '${testdrive.temp-dir}/bad-text.csv'
-  FORMAT CSV WITH HEADER
+  FROM S3 DISCOVER OBJECTS MATCHING 'bad-text.csv' USING BUCKET SCAN 'testdrive-test-${testdrive.seed}'
+  WITH (
+    region = '${testdrive.aws-region}',
+    endpoint = '${testdrive.aws-endpoint}',
+    access_key_id = '${testdrive.aws-access-key-id}',
+    secret_access_key = '${testdrive.aws-secret-access-key}',
+    token = '${testdrive.aws-token}'
+  )
+  FORMAT CSV WITH HEADER (dollars, category)
 
 ! SELECT * FROM bad_text_csv
 contains:Decode error: Text: CSV error at record number 2: invalid UTF-8
 
-# CSV file with comma in header
-
-$ file-append path=header-delimiter.csv
-"interesting,id",name
-1,blat
-
-> CREATE MATERIALIZED SOURCE header_delimited
-  FROM FILE '${testdrive.temp-dir}/header-delimiter.csv'
-  FORMAT CSV WITH HEADER
-
-> SELECT * FROM header_delimited
-interesting,id name mz_line_no
-------------------------------
-1 blat 1
-
 # Declare a key constraint (PRIMARY KEY NOT ENFORCED)
 
 $ kafka-create-topic topic=static-csv-pkne-sink
 
 > CREATE MATERIALIZED SOURCE static_csv_pkne (PRIMARY KEY (zip) NOT ENFORCED)
-  FROM FILE '${testdrive.temp-dir}/static.csv'
-  FORMAT CSV WITH HEADER
```

Comment:
it sounds like we don't support bare `WITH HEADER` anymore, ostensibly because we don't read from s3 in purification...we will need to make sure we remove this being even parseable when we remove file sources from the code side

Reply:
💯